File: 1/2021.07.03/1448.Count Good Nodes In Binary Tree/java/solution.java
class Solution {

    private int res = 0;

    public int goodNodes(TreeNode root) {
        preorder(root, Integer.MIN_VALUE);
        return res;
    }

    private void preorder(TreeNode root, int max) {
        if (root != null) {
            if (root.val >= max) {
                res++;
                max = root.val;
            }
            preorder(root.left, max);
            preorder(root.right, max);
        }
    }
}

File: 1/2021.07.03/1448.Count Good Nodes In Binary Tree/python/solution.py
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def preorder(root, max_val=float('-inf')):
            if root:
                if root.val >= max_val:
                    self.res += 1
                    max_val = root.val
                preorder(root.left, max_val)
                preorder(root.right, max_val)
        
        self.res = 0
        preorder(root)
        return self.res
